### ISSUE TITLE
Error opening magnet links with xdg-open

### ISSUE THREAD

**Issue by zevdg** [transmission-remote-gtk/transmission-remote-gtk#59]
magnet link attempts to open as a .torrent file.  This of course fails with the error "invalid or corrupt torrent file".  Magnet links should be detected and use "add from url" automatically.

**Comment by uhop** [comment]
Yep, the same for me. It appears that this functionality was broken recently.

**Comment by mgedmin** [comment]
This is possibly because xdg-open (or, rather, `gio open` which is spawned by xdg-open on GNOME desktops) rewrites the `magnet:?...` URL to `magnet:///?...` before passing it to `transmission-remote-gtk`.

**Comment by mgedmin** [comment]
I've filed a glib bug: https://bugzilla.gnome.org/show_bug.cgi?id=793014.

I think it would be a good idea to add a workaround in transmission-remote-gtk and rewrite `magnet:///` to `magnet:` as a workaround until that bug's fixed (assuming it's accepted as a valid bug).

**Comment by troxor** [comment]
This was fixed upstream, gio-2.56.2 on Gentoo works great.
